modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java

```diff
@@ -245,17 +246,27 @@ public int produce(final ContentDecoder decoder) throws IOException {
         lock.lock();
         try {
             setInputMode(buffer);
+            int totalRead = 0;
```

Comment:
shall we rename this variable to totalBytesRead?

Reply:
Fixed in 1a225662c71267f47edbbee23cb3dcf7c2106fd3